File: Go/problem005.go
package projecteuler

// Problem 5 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=5
import (
	"log"
	"projecteuler/my"
)

func P005() {
	ans := 1
	for i := 2; i <= 20; i++ {
		ans = my.Lcm(ans, i)
	}
	log.Println(ans)
}

File: Zig/problem005.zig
// zig version 0.13.0
const std = @import("std");
const stdout = std.io.getStdOut().writer();

// #5 Smallest multiple - Project Euler
// http://projecteuler.net/index.php?section=problems&id=5
pub fn main() !void {
    var ans: i64 = 1;
    inline for (1..20 + 1) |i| {
        ans = lcm(ans, i);
    }

    try stdout.print("{}\n", .{ans});
}

fn gcd(x: i64, y: i64) i64 {
    if (y == 0) {
        return x;
    }
    return gcd(y, @mod(x, y));
}

fn lcm(x: i64, y: i64) i64 {
    return @divTrunc(x * y, gcd(x, y));
}
